core/src/main/java/org/infinispan/marshall/protostream/impl/MarshallableUserObject.java

```
@@ -0,0 +1,144 @@
+package org.infinispan.marshall.protostream.impl;
+
+import java.io.IOException;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.marshall.MarshallingException;
+import org.infinispan.commons.marshall.ProtoStreamTypeIds;
+import org.infinispan.commons.util.Util;
+import org.infinispan.protostream.ImmutableSerializationContext;
+import org.infinispan.protostream.RawProtoStreamReader;
+import org.infinispan.protostream.RawProtoStreamWriter;
+import org.infinispan.protostream.RawProtobufMarshaller;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoTypeId;
+import org.infinispan.protostream.impl.WireFormat;
+
+/**
+ * A wrapper message used by ProtoStream Marshallers to allow user objects to be marshalled/unmarshalled by the configured
+ * user marshaller via a ProtoStream Marshaller instance.
+ *
+ * @author Ryan Emerson
+ * @since 10.0
+ */
+@ProtoTypeId(ProtoStreamTypeIds.MARSHALLABLE_USER_OBJECT)
+public class MarshallableUserObject<T> {
+
+   private final T object;
+
+   @ProtoFactory
+   MarshallableUserObject(byte[] bytes) {
+      // no-op never actually used, as we override the default marshaller
```

Maybe we should throw an exception then?